Documentation/Concepts/events.md

```diff
@@ -0,0 +1,133 @@
+
+This is common in accounting, for example:
+Sally adds 100$ into her bank, which would result in an event like "Add 100$ to Sally's account". But if the bank accidentally adds 1000$ instead of the 100$ then a correcting event should be played, like "Subtract 900$ from Sally's account". And with event sourcing, this information is preserved in the event store for eg. later auditing purposes.
+
+### Naming
+To indicate that the event “has happened in the past”, it should be named as a verb in the past tense.  Often it can contain the name of the entity that the change or action is affecting.
+
+- ✅ `DishPrepared`
+- ✅ `ItemAddedToCart`
+- ❌ `StartCooking`
+- ❌ `AddItemToCart`
+
+## “within your system”
+An event represents something interesting that you wish to capture in your system. Instead of seeing state changes and actions as side effects, they are explicitly modeled within the system and captured within the name, state and shape of our Event.
+
+> State transitions are an important part of our problem space and should be modeled within our domain — Greg Young
+
+### Naming
+An event should be expressed in language that makes sense in the domain, also known as [Ubiquitous Language](https://martinfowler.com/bliki/UbiquitousLanguage.html). You should avoid overly technical/CRUD-like events where such terms are not used in the domain.
+
+For example, in the domain of opening up the kitchen for the day and adding a new item to the menu:
+- ✅ `KitchenOpened`
+- ✅ `DishAddedToMenu`
+- ❌ `TakeoutServerReady`
+- ❌ `MenuListingElementUpdated`
+
+## Main structure of an Event
+This is a simplified structure of the main parts of an event.
+
+```csharp
+Event {
+    Content object
+    EventSourceId Guid
+    EventType {
+        EventTypeId Guid
+        Generation int
+    }
+    Public bool
+}
+```
+
+For the whole structure of an event as defined in the gRPC, please check [Contracts](https://github.com/dolittle/Contracts/tree/master/Source/Runtime/Events).
+
+### Content
+This is the object the user commits.
+
+### EventSourceId
+`EventSourceId` represents the source of the event like a "primary key" in a traditional database. By default, [partitioned event handlers]({{< ref "event_handlers_and_filters#event-handlers" >}}) use it as the `PartitionId`.
+
+### EventType
+An `EventType` is a combination of an `EventTypeId` to uniquely identify the type of event it is and the event types `Generation`.
+This decouples the event from a programming language and enables the renaming of events as the domain language evolves.
+Simply put, event types are a wrapper for the actual type of your event. They mask the underlying language-specific type implementation and conventions.
```

Comment:
The first sentence was, to me, a bit confusing. I think that we could just drop these two sentences as it seems to me as that it basically says the same thing as the sentence above
```suggestion
```